### PR TITLE
Illia rudniev/feat/react UI lib

### PR DESCRIPTION
**Pull request by chesterkmr**
### Description
- Prepared ground for react `ui` lib
- added story book
- added few components to for tests

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: packages/ui/.eslintrc.cjs
Comment (Omri-Levy) [inline packages/ui/.eslintrc.cjs:1]: Please use the ESLint file from `packages/config` as the base for Prettier and ESLint.

File: packages/ui/jest.config.ts
Comment (Omri-Levy) [inline packages/ui/jest.config.ts:1]: Please replace Jest with Vitest + testing library. :)

File: packages/ui/package.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+  "name": "@ballerine/ui",
+  "private": true,
```
Comment (Omri-Levy) [inline packages/ui/package.json:3]: Change `private` to `false`

File: packages/ui/package.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  "exports": {
+    ".": {
+      "import": "./lib/index.js",
+      "types": "./lib/index.d.ts"
+    },
+    "./components/*": "./lib/components/*",
+    "./lib/style.css": "./lib/style.css"
+  },
+  "scripts": {
+    "dev": "vite build --watch",
```
Comment (Omri-Levy) [inline packages/ui/package.json:17]: Is `"dev": "vite build --watch"` supposed to be `"dev": "vite"`?

File: packages/ui/package.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  "dependencies": {
+    "@radix-ui/react-dialog": "^1.0.2",
+    "@radix-ui/react-dropdown-menu": "^2.0.5",
+    "@radix-ui/react-label": "^2.0.1",
+    "@radix-ui/react-slot": "^1.0.1",
+    "@tanstack/react-table": "^8.9.2",
+    "@types/classnames": "^2.3.1",
+    "@types/lodash": "^4.14.191",
+    "class-variance-authority": "^0.6.0",
+    "classnames": "^2.3.2",
```
Comment (Omri-Levy) [inline packages/ui/package.json:34]: Replace with `clsx`. :pray:

File: packages/ui/package.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    "dayjs": "^1.11.6",
+    "lodash": "^4.17.21",
+    "lucide-react": "^0.144.0",
+    "react-custom-scrollbars": "^4.2.1",
```
Comment (Omri-Levy) [inline packages/ui/package.json:38]: Check out https://ui.shadcn.com/docs/components/scroll-area

File: packages/ui/package.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    "build-storybook": "storybook build",
+    "test": "NODE_ENV=test jest"
+  },
+  "dependencies": {
+    "@radix-ui/react-dialog": "^1.0.2",
+    "@radix-ui/react-dropdown-menu": "^2.0.5",
+    "@radix-ui/react-label": "^2.0.1",
+    "@radix-ui/react-slot": "^1.0.1",
+    "@tanstack/react-table": "^8.9.2",
+    "@types/classnames": "^2.3.1",
```
Comment (Omri-Levy) [inline packages/ui/package.json:31]: Move `@types/*` deps to `devDependencies`. :)

File: packages/ui/package.json [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    "class-variance-authority": "^0.6.0",
+    "classnames": "^2.3.2",
+    "dayjs": "^1.11.6",
+    "lodash": "^4.17.21",
+    "lucide-react": "^0.144.0",
+    "react-custom-scrollbars": "^4.2.1",
+    "react-json-view": "^1.21.3"
+  },
+  "devDependencies": {
+    "@nx/react": "^16.5.0",
```
Comment (Omri-Levy) [inline packages/ui/package.json:42]: What is `@nx/react` used for here?

File: packages/ui/src/common/enums/workflow-health-status.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export enum WorkflowHealthStatus {
```
Comment (Omri-Levy) [inline packages/ui/src/common/enums/workflow-health-status.ts:1]: Prefer `as const` over enums https://www.youtube.com/watch?v=jjMbPt_H3RQ

File: packages/ui/src/components/atoms/Dialog/Dialog.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  <div
+    className={classnames('flex flex-col space-y-1.5 text-center sm:text-left', className)}
+    {...props}
+  />
+);
+DialogHeader.displayName = 'DialogHeader';
+
+const DialogFooter = ({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) => (
+  <div
+    className={classnames(
```
Comment (Omri-Levy) [inline packages/ui/src/components/atoms/Dialog/Dialog.tsx:70]: Note these instances of `classnames` need to use ShadCN's `cn` utility - you can copy ours from here https://github.com/ballerine-io/ballerine/blob/dev/apps/backoffice-v2/src/common/utils/ctw/ctw.ts

File: packages/ui/src/components/atoms/Dialog/Dialog.tsx
Comment (Omri-Levy) [inline packages/ui/src/components/atoms/Dialog/Dialog.tsx:1]: For files which export many components see how we've approached it in the backoffice https://github.com/ballerine-io/ballerine/tree/dev/apps/backoffice-v2/src/common/components/molecules/DropdownMenu

File: packages/ui/src/components/atoms/Table/Table.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+import classnames from 'classnames';
+
+export const Table = React.forwardRef<HTMLTableElement, React.HTMLAttributes<HTMLTableElement>>(
+  ({ className, ...props }, ref) => (
+    <table
+      ref={ref}
+      className={classnames('caption-bottom w-full text-sm', className)}
+      {...props}
```
Comment (Omri-Levy) [inline packages/ui/src/components/atoms/Table/Table.tsx:9]: Let's use `{children}` explicitly for similar components. :grin:

File: packages/ui/src/components/organisms/WorkflowsTable/WorkflowsTable.tsx [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+        table.setSorting(newSortingValue);
+      } else {
+        const sortingState = updater;
+        onSort(sortingState[0].id, sortingState[0].desc ? 'desc' : 'asc');
+      }
+    },
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  const isEmpty = !items.length && !isFetching;
```
Comment (Omri-Levy) [inline packages/ui/src/components/organisms/WorkflowsTable/WorkflowsTable.tsx:68]: Should `isFetching` be `isLoading` instead?

File: packages/ui/src/components/organisms/WorkflowsTable/WorkflowsTable.tsx [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+interface Props {
+  items: WorkflowTableItem[];
+  sorting?: WorkflowsTableSorting;
+  isFetching?: boolean;
+  columns?: InputColumn[];
+  onSort: (key: string, direction: 'asc' | 'desc') => void;
+}
+
+export function WorkflowsTable({ items, isFetching, sorting, columns, onSort }: Props) {
```
Comment (Omri-Levy) [inline packages/ui/src/components/organisms/WorkflowsTable/WorkflowsTable.tsx:25]: Is there room for a generic table component and `WorkflowsTable` could be built on top of it?

File: packages/ui/src/components/organisms/WorkflowsTable/utils/calculate-hour-difference.ts [hunk 12]
```diff
@@ -0,0 +1,8 @@
+export function calculateHourDifference(dateA: Date, dateB: Date) {
```
Comment (Omri-Levy) [inline packages/ui/src/components/organisms/WorkflowsTable/utils/calculate-hour-difference.ts:1]: Can this be calculated using `dayjs`?

File: packages/ui/vite.config.ts [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  build: {
+    outDir: 'lib',
+    lib: {
+      entry: {
+        ...entries,
+        index: './src/index.ts',
+      },
+      formats: ['es'],
+      name: 'react-ui',
```
Comment (Omri-Levy) [inline packages/ui/vite.config.ts:46]: P0 - Rename to `ui`

File: packages/ui/package.json (hunk 2 above)
Comment (chesterkmr) [reply packages/ui/package.json:17]: `vite` won't produce any output in `lib` which is may be needed for runtime development
Though i think dev vite config could be added

File: packages/ui/package.json [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    "@radix-ui/react-label": "^2.0.1",
+    "@radix-ui/react-scroll-area": "^1.0.2",
+    "@radix-ui/react-slot": "^1.0.1",
+    "@tanstack/react-table": "^8.9.2",
+    "class-variance-authority": "^0.6.0",
+    "clsx": "^1.2.1",
+    "dayjs": "^1.11.6",
+    "lodash": "^4.17.21",
+    "lucide-react": "^0.144.0",
+    "react-custom-scrollbars": "^4.2.1",
```
Comment (Omri-Levy) [inline packages/ui/package.json:37]: Remove `react-custom-scrollbars` if it's no longer in use. :slightly_smiling_face:

File: packages/ui/package.json [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.61.0",
+    "@typescript-eslint/parser": "^5.61.0",
+    "@vitejs/plugin-react": "^4.0.1",
+    "autoprefixer": "^10.4.14",
+    "eslint": "^8.44.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-react-refresh": "^0.4.1",
+    "eslint-plugin-storybook": "^0.6.6",
+    "fast-glob": "^3.3.0",
+    "jest": "^29.6.1",
```
Comment (Omri-Levy) [inline packages/ui/package.json:66]: Remove `jest`.

File: packages/ui/package.json [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+    "@storybook/blocks": "^7.0.26",
+    "@storybook/builder-vite": "^7.0.26",
+    "@storybook/react": "^7.0.26",
+    "@storybook/react-vite": "^7.0.26",
+    "@storybook/testing-library": "^0.0.14-next.2",
+    "@tailwindcss/line-clamp": "^0.4.4",
+    "@types/lodash": "^4.14.191",
+    "@types/node": "^20.4.1",
+    "@types/react": "^18.0.14",
+    "@types/react-custom-scrollbars": "^4.0.10",
```
Comment (Omri-Levy) [inline packages/ui/package.json:55]: Remove `@types/react-custom-scrollbars` if it's no longer in use.

File: packages/ui/package.json [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+    "storybook": "^7.0.26",
+    "tailwindcss": "^3.3.2",
+    "tailwindcss-animate": "^1.0.5",
+    "typescript": "^4.9.5",
+    "vite": "^4.4.0",
+    "vite-plugin-dts": "^1.6.6",
+    "vitest": "^0.33.0"
+  },
+  "overrides": {
+    "@types/react": "18.0.29"
```
Comment (Omri-Levy) [inline packages/ui/package.json:79]: Let's make sure if the real issue was `pnpm-lock.yaml`.

**Review by Omri-Levy — APPROVED** [review]

**Review by alonp99 — APPROVED** [review]

**Pull request merged**